src/hqdn3d.cpp

```diff
@@ -20,48 +20,48 @@
     Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */
 
+#include <cmath>
+#include <cstdlib>
 #include <cstdint>
 #include <algorithm>
 
-#include <vapoursynth/VapourSynth.h>
-#include <vapoursynth/VSHelper.h>
+#include <VapourSynth.h>
+#include <VSHelper.h>
 
 typedef struct Hqdn3dData {
     VSNodeRef *clip;
     const VSVideoInfo *vi;
 
-    double lumSpac   = -1.0;
-    double chromSpac = -1.0;
-    double lumTmp    = -1.0;
-    double chromTmp  = -1.0;
-    int restartLap   = -1;
+    double lumSpac;
+    double chromSpac;
+    double lumTmp;
+    double chromTmp;
+    int restartLap;
 
     int coefs[4][512*16];
-    int planeCnt     = -1;
-    unsigned int ** prevFrame = nullptr;
-    unsigned int ** prevLine  = nullptr;
+    unsigned int *prevFrame[3] = { 0 };
+    unsigned int *prevLine[3] = { 0 };
+    bool process[3];
+    int last_frame = -1; // the last frame returned by hqdn3d
 } Hqdn3dData;
 
 static void
 VS_CC hqdn3dFree(void *instanceData, VSCore *core, const VSAPI *vsapi) {
     (void)core;
 
     Hqdn3dData *d = (Hqdn3dData *)instanceData;
-    if (!d) {
```

Comment:
Even if this should not happen, this will guarantee some safety

Reply:
If you're getting a null pointer here, something is wrong. I don't think hiding the problem is helpful.